titus-common/src/main/java/com/netflix/titus/common/util/ExecutorsExt.java

```diff
@@ -32,4 +43,44 @@ public static ExecutorService namedSingleThreadExecutor(String name) {
         });
     }
 
+    /**
+     * Unbounded elastic thread pool that grows and shrinks as necessary.
+     */
+    public static ExecutorService instrumentedCachedThreadPool(Registry registry, String name) {
+        ThreadFactory threadFactory = new ThreadFactoryBuilder()
+                .setNameFormat(name + "-%d")
+                .setDaemon(true)
+                .build();
+        // similar to Executors.newCachedThreadPool(), but explicitly use the concrete ThreadPoolExecutor type to ensure
+        // it can be instrumented by Spectator
+        ThreadPoolExecutor executor = new ThreadPoolExecutor(
+                0, Integer.MAX_VALUE,
+                60L, TimeUnit.SECONDS,
+                new SynchronousQueue<>(),
+                threadFactory
+        );
+        ThreadPoolMonitor.attach(registry, executor, name);
+        return executor;
+    }
+
+    /**
+     * Fixed size thread pool that pre-allocates all threads. When no more threads are available, requests will block.
+     */
+    public static ExecutorService instrumentedFixedSizeThreadPool(Registry registry, String name, int size) {
+        ThreadFactory threadFactory = new ThreadFactoryBuilder()
+                .setNameFormat(name + "-%d")
+                .setDaemon(true)
+                .build();
+        // similar to Executors.newFixedSizeThreadPool(), but explicitly use the concrete ThreadPoolExecutor type
+        // to ensure it can be instrumented by Spectator
+        ThreadPoolExecutor executor = new ThreadPoolExecutor(
```

Comment:
I see that this is being done to add monitoring, but I would highly recommend using `ForkJoinPool`, which degrades gracefully under load.  When doing a large number of RPCs (Like millons/s), the blocking Queue lock contention begins to serialize the RPC callbacks.  If there is any way to update ThreadPoolMonitor to work with FJP, it would be ideal long term.

Reply:
Yeah, I agree `ForkJoinPool` could be better here. We are not super concerned with the throughput yet, this is not a grpc server where we need to squeeze every possible RPS, so I'm leaning into more instrumentation for now.

Agree it would be nice to have support for an instrumented `ForkJoinPool` in Spectator, we will revisit this as needed in the future.